File: 1406-SubtractTheProductAndSumOfDigitsOfAnInteger/1406-SubtractTheProductAndSumOfDigitsOfAnInteger.js
// Last updated: 3/6/2026, 2:08:59 PM
/**
 * @param {number} n
 * @return {number}
 */
var subtractProductAndSum = function(n) {
    let sum=0;
    let prod=1;
    let temp=n;
    while(temp>0){
        let digit = temp%10;
        sum = sum + digit;
        prod = prod * digit;
        temp = Math.floor(temp/10);
    }
    return prod-sum;
};

File: 1406-SubtractTheProductAndSumOfDigitsOfAnInteger/1406-SubtractTheProductAndSumOfDigitsOfAnInteger.cpp
// Last updated: 7/17/2026, 4:26:31 PM
class Solution {
public:
    int subtractProductAndSum(int n) {
       int product = 1;
       int sum = 0;

       while(n>0){
        int digit = n % 10;
        product*=digit;
        sum+=digit;
        n=n/10;
       }

       return product - sum;
    }
};